### PR TITLE
Add examples to `TyKind::FnDef` and `TyKind::FnPtr` docs

### PR DESCRIPTION
**Pull request by phansch**
This adds two examples to the docs of `TyKind::FnDef` and `TyKind::FnPtr`.

I found these two types a bit confusing when I learned about them and I think adding these examples might help others.

### CODE REVIEW

File: src/librustc/ty/sty.rs [hunk 1]
```diff
@@ -127,9 +127,26 @@ pub enum TyKind<'tcx> {
 
     /// The anonymous type of a function declaration/definition. Each
     /// function has a unique type.
```
Comment (varkor) [inline src/librustc/ty/sty.rs]: Maybe you could also clarify the output syntax here.
```
/// The anonymous type of a function declaration/definition. Each
/// function has a unique type, which is output (for a function 
/// named `foo` returning an `i32`) as `fn() -> i32 {foo}`.
```

File: src/librustc/ty/sty.rs [hunk 2]
```diff
@@ -127,9 +127,26 @@ pub enum TyKind<'tcx> {
...
+    ///
+    /// For example the type of `a` here:
+    ///
+    /// ```rust
+    /// fn foo() -> i32 { 1 }
+    ///
+    /// fn hello() {
```
Comment (varkor) [inline src/librustc/ty/sty.rs]: I think you can get away with just writing it on two lines, which saves a little space:
```rust
fn foo() -> i32 { 1 }
let bar = foo; // bar: fn() -> i32 {foo}
```

**Review by varkor — COMMENTED** [review]
Thanks, I think this improves the readability! Just a couple of small comments to tweak it a little more.

**Comment by phansch** [comment]
Thanks for the review, it should be good now :+1:

**Comment by varkor** [comment]
Thanks!

@bors r+ rollup

**Pull request merged**